### ISSUE TITLE
tvtv.us link gone

### ISSUE THREAD

**Issue by TheDeatCrusher** [iptv-org/epg#403]
What happened to the tvtv.us epg?

**Comment by TheDeatCrusher** [comment]
And also what happened to the rest of the local epg channels?

**Comment by freearhey** [comment]
The API on tvtv.us has recently changed completely and now we need to update the `site_id` of most channels before we can create an EPG. And unfortunately, I do not have time to do this right now.

**Comment by dcole819** [comment]
Is there a alternative epg since tvtv.us is down? I used that because it had so many channels

**Comment by Plugg62208** [comment]
Is it possible to create my own epg with just the channels I need?  I only used tvtv.us for my local channels.

**Comment by dcole819** [comment]
Y’all let me know of a alternative epg that is dependable and has many channels please

**Comment by freearhey** [comment]
@ian10951 Feel free to update the channel list yourself: https://github.com/iptv-org/epg/blob/master/sites/tvtv.us/tvtv.us_us.channels.xml. And the script will load the program during the next daily update.

**Comment by amynkvirani** [comment]
@freearhey could you explain a little on which location data were you pulling these from? For example, for AMCEast and AMCWest, what would you select as the location/zip code and which provider to select (Cox, YouTube etc.)?
Just wanted some information to see if I could help by writing some script to bulk update these.

**Comment by freearhey** [comment]
@amynkvirani You can use https://www.tvtv.us/api/stations and https://www.tvtv.ca/api/stations to get a list of all available channels at once.

**Comment by amynkvirani** [comment]
@freearhey thanks a lot.
making sure I got it right, I should only update the site_id value in your XML, and keep everything else the same so people with already configured EPG wouldn't have to change anything, correct?

**Comment by freearhey** [comment]
@amynkvirani Yes, that's right.

> Also, would you have an older output for https://www.tvtv.us/api/stations which previously worked?

Unfortunately, I don't have it.

**Comment by ohmybahgosh** [comment]
Here's a JSON-->CSV dump of [tvtv/us/api/station](https://www.tvtv.us/api/stations) for anyone who'd have use for it:

[TVTV-STATIONS-API-DUMP.csv](https://github.com/iptv-org/epg/files/7899219/TVTV-STATIONS-API-DUMP.csv)

**Comment by TheDeatCrusher** [comment]
Does anyone have the api for directv epg?

**Comment by freearhey** [comment]
@TheDeatCrusher As far as I know directv.com has no public API, but you can find a complete list of channels in the code of the page https://www.directv.com/guide. And the list is different for each region.

Here, for example, is the list just for New York (zip code: 10001)
https://gist.github.com/freearhey/3ce6294aa1a73c8ead83aa4afc7a24e6

**Comment by TheDeatCrusher** [comment]
Do you have anyway of retreving tvg ids?

**Comment by ian10951** [comment]
Most seem to be working this morning although not all. I do now get a populated xml for both USA and Canada, just about 6 or 7 channels missing that were previously there. Have some channels changed the xml id too ? Greay work anyway.

**Comment by ian10951** [comment]
Much fewer working today, whats going on ? and channels such as History are in Spanish.

**Comment by freearhey** [comment]
@ian10951 could you please specify the specific names of the channels that are missing?

**Comment by ian10951** [comment]
Its easier to show you because a few now just state SIGN OFF as the EPG Data. I cant find anything to upload here so I've sent several Screenshots to an upload site for you to view.

https://easyupload.io/8k2kv9

**Comment by isilrhun** [comment]
I converted the dump @ohmybahgosh uploaded into a channel list, anyone have time to categorize the channels?
[TVTV-STATIONS.xlsx](https://github.com/iptv-org/epg/files/7939124/TVTV-STATIONS.xlsx)

**Comment by isilrhun** [comment]
Here's a list in xmltv format; who wants to put in correct catagories?
[TVTV-STATIONS.csv](https://github.com/iptv-org/epg/files/7955902/TVTV-STATIONS.csv)

**Comment by emjokes** [comment]
Hi everyone - I remapped a bunch of the Canadian channels -- but not all. This was some automatic remapping in Excel, but also a fair bit of manual work. Someone can continue the manual work if you want. 

I am attaching a zip with an xml file containing my changes, as well as an Excel file that contains, in columns A-E the data from the original tvtv.us_ca.channels.xml and columns J-K are IDs and channel names from the tvtv API. I matched column K (tvtv channel name) with Column E (channel name from the xml file in this repo) and where there was a match, I updated the ID (column F). As you will see, there are a lot of N/As in column F that need manual matching. I matched some of them, but don't have any more time for this for the moment.

Can someone (@freearhey, maybe) please merge this? It's far from perfect, but it's much better than what is there right now. I would make a PR myself, but I don't want to close this repo just for one file.
[tvtv.us_ca.channels.xml.zip](https://github.com/iptv-org/epg/files/7966607/tvtv.us_ca.channels.xml.zip)

**Comment by amynkvirani** [comment]
@freearhey so I noticed the EPG wasn't showing data for Comedy Central East. I checked the ID and it does seem to be correct. Any idea what's going wrong?

**Comment by freearhey** [comment]
@amynkvirani Please check again. The channel should now appear.

**Comment by amynkvirani** [comment]
My bad. It wasn't appearing on Tivimate.
Had to clear cache for the app and restart.
All good now, thanks.

**Comment by chamby90** [comment]
> Hi everyone - I remapped a bunch of the Canadian channels -- but not all. This was some automatic remapping in Excel, but also a fair bit of manual work. Someone can continue the manual work if you want.
> 
> I am attaching a zip with an xml file containing my changes, as well as an Excel file that contains, in columns A-E the data from the original tvtv.us_ca.channels.xml and columns J-K are IDs and channel names from the tvtv API. I matched column K (tvtv channel name) with Column E (channel name from the xml file in this repo) and where there was a match, I updated the ID (column F). As you will see, there are a lot of N/As in column F that need manual matching. I matched some of them, but don't have any more time for this for the moment.
> 
> Can someone (@freearhey, maybe) please merge this? It's far from perfect, but it's much better than what is there right now. I would make a PR myself, but I don't want to clone this repo just for one file. [tvtv.us_ca.channels.xml.zip](https://github.com/iptv-org/epg/files/7966607/tvtv.us_ca.channels.xml.zip)

@cad1231 Im not sure I completely understood why you needed to have the excel file updated (since it appears you have all of the updated IDs in the xml file)... but I updated the excel file to match.  Not sure if that is what needed to be done or not, but here is an updated [file.
[Spreadsheet with Index Match_chamby90.xlsx](https://github.com/iptv-org/epg/files/8003953/Spreadsheet.with.Index.Match_chamby90.xlsx)
]

**Comment by amynkvirani** [comment]
> @freearhey so I noticed the EPG wasn't showing data for Comedy Central East. I checked the ID and it does seem to be correct. Any idea what's going wrong?

@freearhey referencing the same channel, this one isn't showing an icon. The link seems to returning a FILE NOT FOUND. How can I fix that? I see there are no icons reference in the updated tvtv.us_us.channels.xml.

**Comment by freearhey** [comment]
@amynkvirani All logos are now loaded from a separate repository [iptv-org/api](https://github.com/iptv-org/api). To replace the logo you just need to update the file: https://github.com/iptv-org/api/blob/master/database/channels.csv

**Comment by Bodyjewels** [comment]
> My bad. It wasn't appearing on Tivimate. Had to clear cache for the app and restart. All good now, thanks.

Trying to get epg working in tivimate, checked spelling not working and advise, newbie Sr

**Comment by freearhey** [comment]
Thanks to @Central302, @Arkina1234, @MichaelRPowell, @RevGear, @m-p-3, @isilrhun, @TheDeatCrusher and @amynkvirani, most of the channels have been returned to the list and are already showing in the guide. Therefore, I think this issue can be closed at this point. 

If you're still missing some channels, feel free to open new [issue](https://github.com/iptv-org/epg/issues/new).

**Comment by magic-ian** [comment]
Where did all the US states and Canadian provinces go though that relied on tvtv? I recall there being many more than just 89 Canadian channels, and many regional US channels.

**Comment by MichaelRPowell** [comment]
@magic-ian if there any specifc ones missing, they can be added back in. A few regional ones for the US channels have been added in. More just need to be added back. [Supposedly there was an old limitation that required the division of files into smaller pieces, but that is no longer a problem for GitHub.](https://github.com/iptv-org/epg/commit/36a277453474b779fedf2aa8a68114e417a51b0e#commitcomment-67689198)

**Comment by magic-ian** [comment]
Thanks @MichaelRPowell, opened a new issue mentioned above to track as per your suggestion. There are too many to make it worthwhile to list all of the missing ones out.
